src/main/java/com/sarapis/orservice/service/DataExchangeServiceImpl.java

```
@@ -19,157 +19,156 @@
+    Integer page,
+    Integer perPage
+  ) {
+    PageRequest pageable = PageRequest.of(page, perPage);
+    Page<DataExchange> exchangePage = dataExchangeRepository.findDataExchanges(userId, fromDate, toDate, pageable);
+    Page<DataExchangeDTO.Response> dtoPage = exchangePage.map(dataExchangeMapper::toResponseDTO);
+    return PaginationDTO.fromPage(dtoPage);
+  }
+
+  @Override
+  @Transactional(readOnly = true)
+  public DataExchangeDTO.Response getDataExchangeById(String id) {
+    DataExchange dataExchange = dataExchangeRepository.findById(id).orElseThrow();
+    return dataExchangeMapper.toResponseDTO(dataExchange);
+  }
+
+  @Override
+  @Transactional
+  public DataExchangeDTO.Response createDataExchange(
+    DataExchangeType type,
+    DataExchangeFormat format,
+    boolean success,
+    String errorMessage,
+    Long size,
+    String userId
+  ) {
+    DataExchange dataExchange = new DataExchange();
+    dataExchange.setId(UUID.randomUUID().toString());
+    dataExchange.setTimestamp(LocalDateTime.now());
+    dataExchange.setType(type);
+    dataExchange.setSuccess(success);
+    dataExchange.setErrorMessage(errorMessage);
+    dataExchange.setFormat(format);
+    dataExchange.setSize(size);
+    dataExchange.setUserId(userId);
+    dataExchange.setDataExchangeFiles(new ArrayList<>());
+
+    DataExchange savedDataExchange = dataExchangeRepository.save(dataExchange);
+    return dataExchangeMapper.toResponseDTO(savedDataExchange);
+  }
+
+  @Override
+  @Transactional
+  public int exportFile(HttpServletResponse response, DataExchangeDTO.Request requestDto) {
+    try {
+      Map<String, Exchangeable> exportMappings = createExportMappings();
+
+      ZipOutputStream zipOutputStream = new ZipOutputStream(response.getOutputStream());
+
+      List<DataExchangeFileDTO.ExchangeFileData> fileDataList = new ArrayList<>();
+      for (DataExchangeDTO.ExchangeableFile file : requestDto.getFiles()) {
+        switch (requestDto.getFormat()) {
```

As a `default` case we should throw an exception `File Type not supported`